src/index.js

```diff
@@ -88,7 +88,7 @@ function getParent (element) {
 
 function getFirstMatchingAncestor (element, nodes) {
   let ancestor = getParent(element)
-  while (ancestor) {
+  while (ancestor && ancestor.nodeType === 1) {
```

Comment:
Guess we may as well check for `nodeType` here; it's pretty useless to try to run the matching algorithm on `document` or `shadowRoot`.

Reply:
Actually just realized this is useless because `getParent` never returns anything but an element. I'll remove it